### PR TITLE
(QA-3346) Automate perf setup for PE/Gatling

### PR DESCRIPTION
**Pull request by samwoods1**
Added new install_gatling directory so that we can preserve the old workflows until we have completely migrated everything into the new workflow.  We still need support for Clamps and FOSS/Gatling.

### CODE REVIEW

File: setup/install_gatling/30_post_install/50_configure_permissive_server_auth.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+test_name 'Configure permissive auth on Puppet Server'
+
+step 'Configure permissive auth.conf on master' do
+  supports_tk_auth = ENV['PUPPET_SERVER_TK_AUTH'] == 'true'
+
+  if supports_tk_auth
+    Beaker::Log.notify "Server supports tk auth, configuring."
+
+    auth_conf = '/etc/puppetlabs/puppetserver/conf.d/auth.conf'
+    create_remote_file(master, auth_conf, <<-EOF)
```
Comment (doug-rosser) [inline setup/install_gatling/30_post_install/50_configure_permissive_server_auth.rb]: In my gut, I think we should at least back up the original file, but this is essentially code from other parts of the project I am willing to :+1: now and submit a PR later.

File: setup/install_gatling/30_post_install/50_configure_permissive_server_auth.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    auth_conf = '/etc/puppetlabs/puppet/auth.conf'
+    create_remote_file(master, auth_conf, <<-EOF)
+path /
+auth any
+allow *
+    EOF
+
+  end
+
```
Comment (doug-rosser) [inline setup/install_gatling/30_post_install/50_configure_permissive_server_auth.rb]: Puppet Server needs to be restarted for the auth.conf change to take effect.

File: setup/install_gatling/30_post_install/50_configure_permissive_server_auth.rb (hunk 1 above)
Comment (doug-rosser) [reply setup/install_gatling/30_post_install/50_configure_permissive_server_auth.rb]: Looks like we are restarting puppet server in https://github.com/puppetlabs/gatling-puppet-load-test/pull/141/files#diff-5c5ae9d30fb908b001bb997c74792edb but as this comes before this test, the auth.conf changes still won't take effect.

File: setup/install_gatling/30_post_install/70_disable_firewall.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+step "disable firewall" do
+  on(master, "puppet resource service firewalld ensure=stopped")
+  on(master, "puppet resource service firewalld ensure=disabled")
```
Comment (doug-rosser) [inline setup/install_gatling/30_post_install/70_disable_firewall.rb]: Beaker **should** have built-in functionality to manage firewalls generically. This is very el7 specific code, which is a pain when dealing with Amazon Linux, which is el6 based.

File: setup/install_gatling/30_post_install/50_configure_permissive_server_auth.rb (hunk 1 above)
Comment (samwoods1) [reply setup/install_gatling/30_post_install/50_configure_permissive_server_auth.rb]: Good call, I re-named the tests to run in a different order.

**Comment by RandellP** [comment]
Test run failed...
```
Begin setup/install_gatling/30_post_install/20_collect_facter_and_scenario_data.rb
Gathering facter data and processing data...
RuntimeError: PUPPET_GATLING_SIMULATION_CONFIG scenario file must be defined
/Users/randell/git/gatling-puppet-load-test/setup/helpers/gatling_config_helper.rb:43
/Users/randell/git/gatling-puppet-load-test/setup/install_gatling/30_post_install/20_collect_facter_and_scenario_data.rb:55
/Users/randell/git/gatling-puppet-load-test/.bundle/gems/beaker-3.29.0/lib/beaker/test_case.rb:133
/Users/randell/git/gatling-puppet-load-test/.bundle/gems/beaker-3.29.0/lib/beaker/test_case.rb:133
/Users/randell/.rvm/rubies/ruby-2.3.0/lib/ruby/2.3.0/benchmark.rb:308
```

**Comment by samwoods1** [comment]
@RandellP I just removed this file completely.  It had a bunch of stuff that should happen after the gatling run was completed.  We can include it in a subsequent check-in after the we automate the gatling record/playback.

**Comment by RandellP** [comment]
still failing...
```
2017-12-08 17:08:12,449 - [Error]: /Stage[main]/Puppet_enterprise::Master::Puppetserver/Puppet_enterprise::Trapperkeeper::Pe_service[puppetserver]/Service[pe-puppetserver]: Failed to call refresh: Systemd restart for pe-puppetserver failed!
        journalctl log for pe-puppetserver:
        -- Logs begin at Fri 2017-12-08 16:41:28 UTC, end at Fri 2017-12-08 17:08:12 UTC. --
        Dec 08 17:07:36 ec2-52-40-235-31.us-west-2.compute.amazonaws.com systemd[1]: Starting pe-puppetserver Service...
        Dec 08 17:08:02 ec2-52-40-235-31.us-west-2.compute.amazonaws.com systemd[1]: Started pe-puppetserver Service.
        Dec 08 17:08:02 ec2-52-40-235-31.us-west-2.compute.amazonaws.com systemd[1]: Reloading pe-puppetserver Service.
        Dec 08 17:08:12 ec2-52-40-235-31.us-west-2.compute.amazonaws.com systemd[1]: pe-puppetserver.service: main process exited, code=exited, status=129/n/a
        Dec 08 17:08:12 ec2-52-40-235-31.us-west-2.compute.amazonaws.com puppetserver[5249]: Process 5008 exited before reload had completed
        Dec 08 17:08:12 ec2-52-40-235-31.us-west-2.compute.amazonaws.com systemd[1]: pe-puppetserver.service: control process exited, code=exited status=1
        Dec 08 17:08:12 ec2-52-40-235-31.us-west-2.compute.amazonaws.com systemd[1]: Reload failed for pe-puppetserver Service.
        Dec 08 17:08:12 ec2-52-40-235-31.us-west-2.compute.amazonaws.com systemd[1]: Unit pe-puppetserver.service entered failed state.
        2017-12-08 17:08:12,449 - [Error]: /Stage[main]/Puppet_enterprise::Master::Puppetserver/Puppet_enterprise::Trapperkeeper::Pe_service[puppetserver]/Service[pe-puppetserver]: Systemd restart for pe-puppetserver failed!
        journalctl log for pe-puppetserver:
        -- Logs begin at Fri 2017-12-08 16:41:28 UTC, end at Fri 2017-12-08 17:08:12 UTC. --
        Dec 08 17:07:36 ec2-52-40-235-31.us-west-2.compute.amazonaws.com systemd[1]: Starting pe-puppetserver Service...
        Dec 08 17:08:02 ec2-52-40-235-31.us-west-2.compute.amazonaws.com systemd[1]: Started pe-puppetserver Service.
        Dec 08 17:08:02 ec2-52-40-235-31.us-west-2.compute.amazonaws.com systemd[1]: Reloading pe-puppetserver Service.
        Dec 08 17:08:12 ec2-52-40-235-31.us-west-2.compute.amazonaws.com systemd[1]: pe-puppetserver.service: main process exited, code=exited, status=129/n/a
        Dec 08 17:08:12 ec2-52-40-235-31.us-west-2.compute.amazonaws.com puppetserver[5249]: Process 5008 exited before reload had completed
        Dec 08 17:08:12 ec2-52-40-235-31.us-west-2.compute.amazonaws.com systemd[1]: pe-puppetserver.service: control process exited, code=exited status=1
        Dec 08 17:08:12 ec2-52-40-235-31.us-west-2.compute.amazonaws.com systemd[1]: Reload failed for pe-puppetserver Service.
        Dec 08 17:08:12 ec2-52-40-235-31.us-west-2.compute.amazonaws.com systemd[1]: Unit pe-puppetserver.service entered failed state.
        2017-12-08 17:08:12,454 - [Notice]: /Stage[main]/Puppet_enterprise::Profile::Orchestrator/Puppet_enterprise::Trapperkeeper::Orchestrator[orchestration-services]/Pe_puppet_authorization::Rule[dispatch: allow pcp-brokers]/Pe_puppet_authorization_hocon_rule[rule-dispatch: allow pcp-brokers]/ensure: created
```

**Comment by RandellP** [comment]
Only thing I found in the puppetserver.log on the MoM was this.
```
2017-12-08 21:58:12,357 WARN  [qtp143093089-290] [puppetserver] Puppet GettextSetup initialization for puppet_enterprise failed with: undefined method `translation_repositories' for GettextSetup:Module
```

**Comment by samwoods1** [comment]
@RandellP Hmm, that looks like a transient failure...  Try again?  Pull the latest, I did make a couple of other tweaks this morning.

**Comment by samwoods1** [comment]
update jira...

**Pull request merged**